src/hotkey/key_window.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <stdint.h>
+
+/*
+ * The key_window module provides a FIFO key_window interface to help with bookkeeping the
```

Comment:
If we can preallocate the entire window when initialize, can we just do that, treat the whole thing as a big array, use offset to traverse the structure and forget about the link list?